src/lib/lspace/layout/lreq.rs

```diff
@@ -0,0 +1,967 @@
+pub fn fast_max<T: PartialOrd>(a: T, b: T) -> T {
+    if a > b {
+        return a;
+    }
+    else {
+        return b;
+    }
+}
+
+pub fn fast_maxopt<T: PartialOrd>(a: Option<T>, b: Option<T>) -> Option<T> {
+    match (a, b) {
```

Comment:
Which is the probable common case? If it's the (Some(x), Some(y)) case, that's probably worth putting first as (I think) pattern matching is tried in order.